mvs/common/adapter_base.py

```diff
@@ -0,0 +1,146 @@
+        )
+
+        for camera, intrinsic in intrinsics.items():
+            subfolder = path_to_preprocessed / camera
+            subfolder.mkdir()
+
+            with open(subfolder / "intrinsic.txt", "w") as intrinsics_file:
+                intrinsics_file.write(
+                    f"{intrinsic.f} {intrinsic.cx} {intrinsic.cy} {intrinsic.k}"
+                )
+
+        raw_images = path_to_raw_images.iterdir()
+        for raw_image in raw_images:
+            raw_image_name = raw_image.name
+
+            match len(raw_image_name), int(raw_image_name[-5]):
+                case 16, 0:
+                    chosen_camera = "a308_0"
+                case 16, 1:
+                    chosen_camera = "a308_1"
+                case 21, 0:
+                    chosen_camera = "a311_0"
+                case 21, 1:
+                    chosen_camera = "a311_1"
+                case _:
+                    raise ValueError("Invalid image name")
+
+            shutil.copyfile(
+                raw_image, path_to_preprocessed / chosen_camera / raw_image_name
+            )
+
+        for cam_subfolder in path_to_preprocessed.iterdir():
+            camera_name = cam_subfolder.name
+
+            images_list = sorted(cam_subfolder.iterdir())[:-1]
+            with open(cam_subfolder / "images_list.txt", "w") as images_list_file:
+                for image in images_list:
+                    images_list_file.write(f"{camera_name}/{image.name}\n")
+
+    def _sparse_reconstruction(self):
+        workdir = Path(self.input_dir)
+
+        database_path = workdir / "database.db"
+        images_path = workdir / "images"
+        sparse_reconstruction_path = workdir / "sparse"
+        dense_reconstruction_path = workdir / "dense"
+        
+        if os.path.exists(sparse_reconstruction_path):
+            shutil.rmtree(sparse_reconstruction_path)
+        if os.path.exists(dense_reconstruction_path):
+            shutil.rmtree(dense_reconstruction_path)
+        sparse_reconstruction_path.mkdir(exist_ok=False)
+        dense_reconstruction_path.mkdir(exist_ok=False)
+
+        feature_extractor = FeatureExtractor(
+            database_path, images_path, init_camera_params=True, one_camera_per_folder=True
+        )
+        feature_matcher = FeatureMatcher(database_path)
+        sparse_reconstructor = SparseReconstructor(
+            database_path,
+            images_path,
+            sparse_reconstruction_path,
+            refine_principal_point=True,
+            tri_min_angle=0.01,
+            ignore_two_view_tracks=False,
+        )
+        undistorter = Undistorter(
+            images_path, sparse_reconstruction_path, dense_reconstruction_path
+        )
+        feature_extractor.calculate_features()
+        feature_matcher.match_features()
+        sparse_reconstructor.get_sparse_reconstruction()
+        undistorter.undistort()
+    
+    @abstractmethod
+    def _convert_colmap_to_mvsnet(self) -> None:
```

Comment:
Why is the return value hint specified only here?

Reply:
I don't think that this method should be in interface of this abstract base class